pkg/util/winutil/eventlog.go

```diff
@@ -0,0 +1,30 @@
+package winutil
+
+import (
+	"golang.org/x/sys/windows/svc/eventlog"
+)
+
+// LogEventViewer will open the event viewer API and log a single message
+// to the event viewer.  The string identified in the msgnum parameter
+// must exist in the application's message catalog
+// go log api only provides for a single argument to be passed, so can
+// only include one positional argument
+func LogEventViewer(servicename string, msgnum uint32, arg string) {
+	elog, err := eventlog.Open(servicename)
+	if err != nil {
+		return
+	}
+	defer elog.Close()
+	switch msgnum & 0xF0000000 {
+	case 0x40000000:
+		// Info level message
+		elog.Info(msgnum, arg)
```

Comment:
Just want to make sure...

is the `msgnum` we want to pass here the full message number, or do we have to mask out the most significant bits?

Reply:
It's the full message number.  The high order 4 bits just indicates if it's info/warning/error level.  So the `case` is just so we send it to the right `log` method.  But, we need the whole message number.